### PR TITLE
[ioctl] Build Read unittest

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Build `Read` unittest for `action` command

refer to #3444 & #3472 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestRead

**Test Configuration**:
- Firmware version: Windows 10 (WSL Version: Ubuntu 18.04)
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LuckyPigeon** [comment]
L41, L320, L334 fix for code smell might get confused easily, should I revert them? The code show as below.
https://github.com/iotexproject/iotex-core/blob/604cc63fcb237908de8abdd6d98ff76a9a8b62b0/ioctl/newcmd/action/action_test.go#L320

File: ioctl/newcmd/action/action_test.go [hunk 1]
```diff
@@ -270,3 +272,70 @@ func TestSendAction(t *testing.T) {
...
+
+	ks := keystore.NewKeyStore(t.TempDir(), 2, 1)
+	acc, err := ks.NewAccount("123456")
+	require.NoError(err)
+	accAddr, err := address.FromBytes(acc.Address.Bytes())
+	require.NoError(err)
```
Comment (huof6829) [inline ioctl/newcmd/action/action_test.go:287]: Just want a account address.  use `accAddr := identityset.Address(0)`  simply

File: ioctl/newcmd/action/action_test.go [hunk 2]
```diff
@@ -270,3 +272,70 @@ func TestSendAction(t *testing.T) {
...
+	chainMetaResponse := &iotexapi.GetChainMetaResponse{ChainMeta: &iotextypes.ChainMeta{}}
+	elp := createEnvelope(0)
+	cost, err := elp.Cost()
+	require.NoError(err)
+	accountResponse := &iotexapi.GetAccountResponse{AccountMeta: &iotextypes.AccountMeta{
+		Address:      accAddr.String(),
+		Nonce:        1,
+		PendingNonce: 1,
+		Balance:      cost.String(),
+	}}
```
Comment (huof6829) [inline ioctl/newcmd/action/action_test.go:298]: not use, delete

File: ioctl/newcmd/action/action_test.go [hunk 3]
```diff
@@ -270,3 +272,70 @@ func TestSendAction(t *testing.T) {
...
+
+	client.EXPECT().SelectTranslation(gomock.Any()).Return("action", config.English).Times(4)
+	client.EXPECT().SetEndpointWithFlag(gomock.Any()).Do(func(_ func(*string, string, string, string)) {
+		// SetEndpointWithFlag take a function as parameter
+	}).Times(2)
+	client.EXPECT().SetInsecureWithFlag(gomock.Any()).Do(func(_ func(*bool, string, bool, string)) {
+		// SetInsecureWithFlag take a function as parameter
+	}).Times(2)
```
Comment (huof6829) [inline ioctl/newcmd/action/action_test.go:306]: delete `.Do(xxx...)` simply

File: ioctl/newcmd/action/action_test.go [hunk 4]
```diff
@@ -270,3 +272,70 @@ func TestSendAction(t *testing.T) {
...
+	client.EXPECT().APIServiceClient().Return(apiServiceClient, nil).Times(2)
+
+	t.Run("reads smart contract on IoTeX blockchain", func(t *testing.T) {
+		client.EXPECT().AddressWithDefaultIfNotExist(gomock.Any()).Return("test", nil)
+		apiServiceClient.EXPECT().GetChainMeta(gomock.Any(), gomock.Any()).Return(chainMetaResponse, nil)
+		apiServiceClient.EXPECT().GetAccount(gomock.Any(), gomock.Any()).Return(accountResponse, nil)
+		apiServiceClient.EXPECT().SendAction(gomock.Any(), gomock.Any()).Return(&iotexapi.SendActionResponse{}, nil)
```
Comment (huof6829) [inline ioctl/newcmd/action/action_test.go:313]: not use, delete

File: ioctl/newcmd/action/action_test.go [hunk 5]
```diff
@@ -270,3 +272,70 @@ func TestSendAction(t *testing.T) {
...
+		apiServiceClient.EXPECT().ReadContract(gomock.Any(), gomock.Any()).Return(&iotexapi.ReadContractResponse{
+			Data: "test",
+		}, nil)
+
+		cmd := NewActionCmd(client)
+		registerSignerFlag(client, cmd)
+		_, err := util.ExecuteCmd(cmd, _signerFlag, "test")
+		require.NoError(err)
+		signer, err := cmd.Flags().GetString(signerFlagLabel)
+		require.NoError(err)
```
Comment (huof6829) [inline ioctl/newcmd/action/action_test.go:323]: delete,  just want singer's value,  use `"test"` simply

File: ioctl/newcmd/action/action_test.go [hunk 6]
```diff
@@ -270,3 +272,70 @@ func TestSendAction(t *testing.T) {
...
+
+	t.Run("failed to get signer address", func(t *testing.T) {
+		expectErr := errors.New("failed to get signer address")
+		client.EXPECT().AddressWithDefaultIfNotExist(gomock.Any()).Return("", expectErr)
+		cmd := NewActionCmd(client)
+		registerSignerFlag(client, cmd)
+		_, err := util.ExecuteCmd(cmd, _signerFlag, "test")
+		require.NoError(err)
+		signer, err := cmd.Flags().GetString(signerFlagLabel)
+		require.NoError(err)
```
Comment (huof6829) [inline ioctl/newcmd/action/action_test.go:337]: the same

File: ioctl/newcmd/action/action_test.go (hunk 5 above)
Comment (huof6829) [reply ioctl/newcmd/action/action_test.go:323]: Delete line#253~255  #266~268  etc..  in `TestSendAction`.
Because just want the value of `passwd`

**Comment by huof6829** [comment]
@LuckyPigeon   I delete the useless codes and variable. so the codes seems clearly.

**Comment by LuckyPigeon** [comment]
@huof6829 
Thanks for the help, I'll do the same change on #3631

**Review by huof6829 — APPROVED** [review]

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**